magento/magento2
# MC-38897: URL Rewrite is lost when adding a existing product to another store view

### Description (*)

The problem was in the process of URL rewrites generation if product is saved in global scope
This PR fixes issue by forcing generation URL rewrites for stores with scope overridden URL key in case if category for store was changed and category url_key is used for URL rewrite for products

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#30658

### Manual testing scenarios (*)

1. Create 2 websites with 2 stores
2. Assign different root categories for each store and create sub category for each
3. Create product
4. Enable Stores->Configuration->Catalog->Catalog->Search Engine Optimization->Use Categories Path for Product URLs configuration
5. Assign product to both websites by copying data from non default values (f.e. from base website store view) so url_key for new website should be non default
6. Add remove categories for the product in new website.
#### Expected result
Url rewrite is generated for categor/product if category is assigned to product and removed if category is removed from product.




### Questions or comments


### Contribution checklist (*)
